docs/Documentation/UserGuide/9-Tools-spark-iotdb.md

```diff
@@ -0,0 +1,141 @@
+|    2 | null                          | false                    | aaa                        | 2.2                           | null                     | null                       |
+|    3 | null                          | null                     | null                       | 2.1                           | true                     | null                       |
+|    4 | null                          | true                     | bbb                        | null                          | null                     | null                       |
+|    5 | null                          | null                     | null                       | null                          | false                    | null                       |
+|    6 | null                          | null                     | ccc                        | null                          | null                     | null                       |
+
+You can also use narrow table form which as follows: (You can see part 4 about how to use narrow form)
+
+| time | device_name                   | status                   | hardware                   | temperature |
+|------|-------------------------------|--------------------------|----------------------------|-------------------------------|
+|    1 | root.ln.wf02.wt01             | true                     | null                       | 2.2                           | 
+|    1 | root.ln.wf02.wt02             | true                     | null                       | null                          | 
+|    2 | root.ln.wf02.wt01             | null                     | null                       | 2.2                          |                 
+|    2 | root.ln.wf02.wt02             | false                    | aaa                        | null                           |                   
+|    3 | root.ln.wf02.wt01             | true                     | null                       | 2.1                           |                 
+|    4 | root.ln.wf02.wt02             | true                     | bbb                        | null                          |                  
+|    5 | root.ln.wf02.wt01             | false                    | null                       | null                          |                   
+|    6 | root.ln.wf02.wt02             | null                     | ccc                        | null                          |                   
+
+# 4. Transform between wide and narrow table
+
+## from wide to narrow
+```
+import org.apache.iotdb.sparkdb._
+
+val wide_df = spark.read.format("org.apache.iotdb.sparkdb").option("url", "jdbc:iotdb://127.0.0.1:6667/").option("sql", "select * from root where time < 1100 and time > 1000").load
+val narrow_df = Transformer.toNarrowForm(spark, wide_df)
+```
+
+## from narrow to wide
+```
+import org.apache.iotdb.sparkdb._
+
+val wide_df = Transformer.toWideForm(spark, narrow_df)
+```
```

Comment:
As @jixuan1989 noted, java docs should be commented in the document too.
So, maybe you can add # 5. Java user guide like:
```
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SparkSession;

public class Tmp {

  public static void main(String[] args) {
    SparkSession spark = SparkSession
        .builder()
        .appName("Build a DataFrame from Scratch")
        .master("local[*]")
        .getOrCreate();

    Dataset<Row> df = spark.read().format("org.apache.iotdb.sparkdb")
        .option("url","jdbc:iotdb://127.0.0.1:6667/")
        .option("sql","select * from root").load();

    df.printSchema();

    df.show();
  }
}

```